torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -5137,7 +5141,9 @@ std::tuple<Tensor, Tensor> householder_product_backward(
   // eg. if both are BatchedTensor at different level.
   if (areAnyTensorSubclassLike({input, tau, K})) {
     // k + 1 if input_grads hold a matrix of zeros for inactive parts of input.
-    auto input_grads = std::vector<Tensor>(k < input.size(-1) ? k + 1 : k);
+    // guard_int is because we need to know a concrete size to allocate
+    auto input_grads = std::vector<Tensor>(
+        k < input.sym_size(-1).guard_int(__FILE__, __LINE__) ? k + 1 : k);
```

Comment:
This guard is **not** needed. I had a different one here before deciding to just guard on k above, so got confused.